stingray/lightcurve.py

```diff
@@ -688,3 +688,17 @@ def read(self, filename, format_='pickle'):
 
         else:
             utils.simon("Format not understood.")
+
+    def split_by_gti(self):
+        """Splits the `LightCurve` into a list of `LightCurve`s , using GTIs."""
```

Comment:
I don't see it like a typical use case. If the user wants to change the GTIs and then split, it is straightforward anyway:
```
lc.gti = mynewwonderfulgtis
split_lcs = lc.split_by_gti()
```